DataFormats/L1Trigger/interface/P2GTCandidate.h

```diff
@@ -0,0 +1,328 @@
+        throw std::invalid_argument("Object doesn't have iso");
+      }
+      return static_cast<int>(hwIso_);
+    }
+
+    hwQual_t hwQual() const {
+      if (!hwQual_) {
+        throw std::invalid_argument("Object doesn't have qual");
+      }
+      return static_cast<int>(hwQual_);
+    }
+
+    hwCharge_t hwCharge() const {
+      if (!hwCharge_) {
+        throw std::invalid_argument("Object doesn't have charge");
+      }
+      return static_cast<int>(hwCharge_);
+    }
+
+    hwD0_t hwD0() const {
+      if (!hwD0_) {
+        throw std::invalid_argument("Object doesn't have d0");
+      }
+      return static_cast<int>(hwD0_);
+    }
+
+    hwBeta_t hwBeta() const {
+      if (!hwBeta_) {
+        throw std::invalid_argument("Object doesn't have beta");
+      }
+      return static_cast<int>(hwBeta_);
+    }
+
+    hwMass_t hwMass() const {
+      if (!hwMass_) {
+        throw std::invalid_argument("Object doesn't have mass");
+      }
+      return static_cast<int>(hwMass_);
+    }
+
+    hwIndex_t hwIndex() const {
+      if (!hwIndex_) {
+        throw std::invalid_argument("Object doesn't have index");
+      }
+      return static_cast<int>(hwIndex_);
+    }
+
+    hwSeed_pT_t hwSeed_pT() const {
+      if (!hwSeed_pT_) {
+        throw std::invalid_argument("Object doesn't have seed_pT");
+      }
+      return static_cast<int>(hwSeed_pT_);
+    }
+
+    hwSeed_z0_t hwSeed_z0() const {
+      if (!hwSeed_z0_) {
+        throw std::invalid_argument("Object doesn't have seed_z0");
+      }
+      return static_cast<int>(hwSeed_z0_);
+    }
+
+    hwSca_sum_t hwSca_sum() const {
+      if (!hwSca_sum_) {
+        throw std::invalid_argument("Object doesn't have sca_sum");
+      }
+      return static_cast<int>(hwSca_sum_);
+    }
+
+    hwNumber_of_tracks_t hwNumber_of_tracks() const {
+      if (!hwNumber_of_tracks_) {
+        throw std::invalid_argument("Object doesn't have number_of_tracks");
+      }
+      return static_cast<int>(hwNumber_of_tracks_);
+    }
+
+    hwSum_pT_pv_t hwSum_pT_pv() const {
+      if (!hwSum_pT_pv_) {
+        throw std::invalid_argument("Object doesn't have sum_pT_pv");
+      }
+      return static_cast<int>(hwSum_pT_pv_);
+    }
+
+    hwType_t hwType() const {
+      if (!hwType_) {
+        throw std::invalid_argument("Object doesn't have type");
+      }
+      return static_cast<int>(hwType_);
+    }
+
+    hwNumber_of_tracks_in_pv_t hwNumber_of_tracks_in_pv() const {
+      if (!hwNumber_of_tracks_in_pv_) {
+        throw std::invalid_argument("Object doesn't have number_of_tracks_in_pv");
+      }
+      return static_cast<int>(hwNumber_of_tracks_in_pv_);
+    }
+
+    hwNumber_of_tracks_not_in_pv_t hwNumber_of_tracks_not_in_pv() const {
+      if (!hwNumber_of_tracks_not_in_pv_) {
+        throw std::invalid_argument("Object doesn't have hwNumber_of_tracks_not_in_pv");
+      }
+      return static_cast<int>(hwNumber_of_tracks_not_in_pv_);
+    }
+
+    double pT() const {
```

Comment:
I have to check. More problematic is that this class is not derived from the standard Candidate class.

Reply:
They need methods pt(), eta(), phi(), mass() which is the case so far because all the different incomming trigger objects in Collections (such as RecoChargedCandidate, L1MuonParticle Collections) are subclassed from within the hierarchy starting with the ```Candidate``` class (which in turn contains a 4-vector used as 4-momentum). Id is added by the filter itself as its interpretation of what the object is. The 4-mon plus id is extracted and packed up in the single trigger object summary collection at HLT.